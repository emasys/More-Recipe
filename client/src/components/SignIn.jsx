import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../actions';

//components
import Navbar from './Navbar';
/**
 *
 *
 * @class SignIn
 * @extends {Component}
 */
class SignIn extends Component {
  /**
   * Creates an instance of SignIn.
   * @param {any} props
   * @memberof SignIn
   */
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      showErrMessage: 'fade'
    };

    this.emailChanged = this.emailChanged.bind(this);
    this.pwChanged = this.pwChanged.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  /**
   *
   *
   * @param {any} e
   * @memberof SignIn
   * @returns {any} email input text
   */
  emailChanged(e) {
    this.setState({
      email: e.target.value
    });
  }
  /**
   *
   *
   * @param {any} e
   * @memberof SignIn
   * @returns {any} password input text
   */
  pwChanged(e) {
    this.setState({
      password: e.target.value
    });
  }
  /**
   *
   *
   * @param {any} e
   * @memberof SignIn
   * @returns {any} a new page
   */
  handleSubmit(e) {
    e.preventDefault();

    this.props.signIn(this.state).then(() => {
      if (this.props.signin.signIn.success) {
        this.setState({
          showErrMessage: 'fade'
        });
        let linkPath = '/';
        if (this.props.match.params) {
          linkPath = this.props.location.pathname;
          if (linkPath === '/signin') linkPath = '/';
        }
        window.location.href = linkPath;
      }
      if (this.props.signin.signIn.success === false) {
        this.setState({
          showErrMessage: 'show'
        });
      }
    });
  }
  /**
   *
   *
   * @returns {any} jsx
   * @memberof SignIn
   */
  render() {
    const { email } = this.state;
    return (
      <section className="container mt-100 mb-100 ">
        <Navbar />
        <div className="row justify-content-center">
          <div className="col-lg-8 col-sm-12 text-center ">
            <img src="../img/logo.png" alt="logo" />
            <p className=" mt-5 text-dark bg-mirror header-title">
              “I hate the notion of a secret recipe. Recipes are by nature
              derivative and meant to be shared that is how they improve, are
              changed, how new ideas are formed. To stop a recipe in it's
              tracks, to label it "secret" just seems mean.” ― Molly Wizenberg
            </p>
          </div>
          <div className="col-lg-4 col-sm-12 ">
            <form
              id="signin"
              className="form-items"
              onSubmit={this.handleSubmit}
            >
              <ul className="form row">
                <li className="col-lg-12 col-sm-12">
                  <label>Email</label>
                  <input
                    type="email"
                    required
                    value={email}
                    onChange={this.emailChanged}
                    className="col-lg-12 col-sm-12"
                    name="email"
                    id="inputEmail"
                    placeholder="example@example.com"
                  />
                </li>
                <li className="col-lg-12 col-sm-12">
                  <label>Password</label>
                  <input
                    type="password"
                    required
                    onChange={this.pwChanged}
                    className="col-lg-12 col-sm-12"
                    name="pass"
                    id="inputPassword"
                    placeholder="**********"
                  />
                  <div
                    className={`text-danger ${this.state.showErrMessage}`}
                    id="moniker_error"
                  >
                    Invalid email or password
                  </div>
                </li>
                <li className="col-lg-12 col-sm-12">
                  <button type="submit" className="btn btn-dark btn-lg ">
                    Sign in
                  </button>&nbsp;
                  <Link to="/signup" className="btn btn-dark btn-lg">
                    Sign up
                  </Link>
                </li>
              </ul>
            </form>
          </div>
        </div>
      </section>
    );
  }
}

const mapStateToProps = state => ({
  signin: state.user
});

export default connect(mapStateToProps, actions)(SignIn);
