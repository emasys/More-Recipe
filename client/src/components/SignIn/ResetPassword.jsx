import React from 'react';
import PropTypes from 'prop-types';

const onFocus = () => {
  document.querySelector('.rubics').classList.add('d-block');
};

const ResetPasswordForm = props => {
  const { recoveryEmail, newPassword } = props.state;
  return (
    <form className="" onSubmit={props.resetPassword}>
      <ul className="form row">
        <li className="col-lg-12 col-sm-12">
          <label>Email</label>
          <input
            type="email"
            required
            value={recoveryEmail}
            onChange={props.onChange}
            className="col-lg-12 col-sm-12"
            name="recoveryEmail"
            id="recoveryEmail"
            placeholder="example@example.com"
          />
          <div className="text-danger" id="recoverEmail_error" />
        </li>
        <li className="col-lg-12 col-sm-12">
          <label>New Password</label>
          <input
            type="password"
            required
            className="col-lg-12 col-sm-12"
            onChange={props.onChange}
            onFocus={onFocus}
            value={newPassword}
            name="newPassword"
            id="newPassword"
            placeholder="min of 8 characters"
          />
          {/* <div className="text-danger" id="newPassword_error" /> */}
          <div className="rubics">
            <p className="text-danger" id="password_error" />
            <p className="alpha">Should contain at least one alphabet</p>
            <p className="numbers" id="check_numbers">
              Should contain at least one number
            </p>
            <p className="characters">
              Should contain at least one special character
            </p>
            <p className="minLength">Should not be less than 8 characters</p>
            <p className="maxLength">
              Should not be greater than 20 characters
            </p>
          </div>
        </li>
        <li className="col-lg-12 col-sm-12">
          <label>Confirm New Password</label>
          <input
            type="password"
            required
            onChange={props.onChange}
            className="col-lg-12 col-sm-12"
            name="confirmPassword"
            id="confirmPassword"
            placeholder="min of 8 characters"
          />
          <div className="text-danger" id="confirmPassword_error" />
        </li>
        <li className="col-lg-12 col-sm-12">
          <label>Token</label>
          <input
            type="text"
            required
            className="col-lg-12 col-sm-12"
            name="token"
            placeholder="enter 4-digit token"
          />
          <div className="text-danger" id="token_error" />
          <button
            className="btn btn-info mt-5 btn-block btn-lg "
            onClick={props.generateToken}
          >
            send token to my mail
          </button>
        </li>
        <li className="col-lg-12 col-sm-12">
          <button type="submit" className="btn btn-dark btn-block btn-lg ">
            Save New Password
          </button>
        </li>
        <li className="col-12 text-center">
          <span className="text-info hovered" onClick={props.resetForm}>
            Go back to sign in
          </span>
        </li>
      </ul>
    </form>
  );
};

export default ResetPasswordForm;
