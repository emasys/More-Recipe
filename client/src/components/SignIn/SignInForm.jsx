import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const SignInForm = props => {
  const {
    email,
    showErrMessage,
  } = props.state;

  return (
    <form id="signin" className="form-ite" onSubmit={props.handleSubmit}>
      <ul className="form row">
        <li className="col-lg-12 col-sm-12">
          <label>Email</label>
          <input
            type="email"
            required
            value={email}
            onChange={props.emailChanged}
            className="col-lg-12 col-sm-12"
            name="email"
            id="inputEmail"
            placeholder="example@example.com"
          />
        </li>
        <li className="col-lg-12 col-sm-12">
          <label>Password</label>
          <input
            type="password"
            required
            onChange={props.pwChanged}
            className="col-lg-12 col-sm-12"
            name="pass"
            id="inputPassword"
            placeholder="**********"
          />
          <div className={`text-danger ${showErrMessage}`} id="moniker_error">
            Invalid email or password
          </div>
        </li>
        <li className="col-lg-12 col-sm-12 text-center">
          <button type="submit" className="btn btn-dark btn-lg col-5">
            Sign in
          </button>&nbsp;
          <Link to="/signup" className="btn btn-dark btn-lg col-5">
            Sign up
          </Link>
        </li>
        <li className="col-lg-12 col-sm-12 text-center">
          <span className="text-info hovered" onClick={props.resetForm}>
            forgot password?
          </span>
        </li>
      </ul>
    </form>
  );
};

export default SignInForm;
