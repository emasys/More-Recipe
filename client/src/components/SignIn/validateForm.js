import { isEqual } from 'lodash';

const validateForm = data => {
  if (data.email === '') {
    document.querySelector('#recoverEmail_error').innerHTML =
      'Please enter a valid email address';
    data.email = null;
  } else {
    document.querySelector('#recoverEmail_error').innerHTML = '';
  }

  if (data.password.length < 8) {
    document.querySelector('#newPassword_error').innerHTML =
      'Please enter a valid password not less than 8 characters';
    data.password = null;
  } else {
    document.querySelector('#newPassword_error').innerHTML = '';
  }

  if (data.token === '') {
    document.querySelector('#token_error').innerHTML =
      'Check your email for your token';
    data.token = null;
  } else {
    document.querySelector('#token_error').innerHTML = '';
  }

  if (!isEqual(data.password, data.confirmPassword)) {
    document.querySelector('#confirmPassword_error').innerHTML =
      'Your password did not match';
    data.confirmPassword = null;
  } else {
    document.querySelector('#confirmPassword_error').innerHTML = '';
  }

  if (data.email && data.confirmPassword && data.password && data.token) {
    return true;
  } else {
    return false;
  }
};

export default validateForm;
