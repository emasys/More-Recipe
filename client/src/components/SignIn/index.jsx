import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import Pace from 'react-pace-progress';

// actions
import {
  resetPassword,
  signIn,
  compareToken,
  sendToken
} from '../../actions/authActions';
import { isLoading } from '../../actions';
//components
import Navbar from '../Navbar';
import SignInForm from './SignInForm';
import ResetPasswordForm from './ResetPassword';
import errorMessages, {
  validateEmail,
  validatePassword,
  confirmPassword
} from '../SignUp/Validators';

/**
 *
 *@param {object} event
 * @class SignIn
 * @extends {Component}
 */
class SignIn extends Component {
  /**
   * Creates an instance of SignIn.
   * @param {any} props
   * @memberof SignIn
   */
  constructor(props) {
    super(props);
    this.state = {
      recoveryEmail: '',
      email: '',
      password: '',
      newPassword: '',
      confirmPassword: '',
      showErrMessage: 'fade',
      message: '',
      success: false,
      showProps: false,
      resetPass: false
    };
  }
  /**
   * @returns {any}
   * invoked immediately after a
   * component is mounted
   * @memberof SignIn
   */
  componentDidMount() {
    if (this.props.msg) {
      this.setState({
        message: this.props.msg,
        showProps: true
      });
    }
  }
  /**
   * @returns {any}
   * invoked before a mounted component
   * receives new props
   * @param {any} nextProps
   * @memberof SignIn
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.signin.signIn) {
      if (nextProps.signin.signIn.success) {
        this.setState({
          showErrMessage: 'fade'
        });
        let linkPath = '/';
        if (nextProps.match.params) {
          linkPath = nextProps.location.pathname;
          if (linkPath === '/signin') linkPath = '/';
        }
        window.location.href = linkPath;
      }
      if (nextProps.signin.signIn.data.success === false) {
        this.setState({
          showErrMessage: 'show'
        });
      }
    }
    if (nextProps.reset) {
      if (nextProps.reset.success) {
        this.setState({
          resetPass: false,
          success: true
        });
      } else {
        document.querySelector('#recoverEmail_error').innerHTML =
          "User account not found, <a href='/signup'>sign up</a>";
      }
    }
  }

  clearError = () => {
    this.setState({
      showErrMessage: 'fade'
    });
  };
  /**
   *
   *
   * @param {any} event
   * @memberof SignIn
   * @returns {any} password input text
   */
  onChange = event => {
    this.setState(
      {
        [event.target.name]: event.target.value
      },
      () => {
        if (this.state.recoveryEmail) {
          validateEmail(
            this.state.recoveryEmail,
            'recoveryEmail',
            'recoveryEmail'
          );
        }
        if (this.state.newPassword) {
          validatePassword(
            this.state.newPassword,
            'newPassword',
            'password_error'
          );
        }
        if (this.state.confirmPassword) {
          confirmPassword(
            this.state.newPassword,
            this.state.confirmPassword,
            'confirmPassword',
            'confirmPassword_error'
          );
        }
      }
    );
  };
  /**
   *
   *
   * @memberof SignIn
   * @returns {any} password input text
   */
  resetForm = () => {
    if (!this.state.resetPass) {
      this.setState({
        resetPass: true,
        success: false
      });
    } else {
      this.setState({
        resetPass: false,
        success: false
      });
    }
  };
  /**
   *
   *
   * @param {any} event
   * @memberof SignIn
   * @returns {any}
   * triggers reset password action
   */
  resetPassword = event => {
    event.preventDefault();
    const data = {
      email: event.target.elements.recoveryEmail.value.trim(),
      password: event.target.elements.newPassword.value,
      confirmPassword: event.target.elements.confirmPassword.value,
      token: event.target.elements.token.value.trim()
    };
    if (errorMessages()) {
      this.props.compareToken(data).then(() => {
        if (this.props.signin.compareToken.success === true) {
          this.props.resetPassword(data);
        } else {
          document.querySelector('#token_error').innerHTML = 'invalid token';
        }
      });
    }
  };

  generateToken = event => {
    event.preventDefault();
    this.props.sendToken({ email: this.state.recoveryEmail });
  };
  /**
   *
   *
   * @param {any} event
   * @memberof SignIn
   * @returns {any} a new page
   */
  handleSubmit = event => {
    event.preventDefault();
    const data ={
      email: event.target.elements.email.value.trim(),
      password: event.target.elements.pass.value
    }
    this.props.signIn(data);
  };
  /**
   *
   *
   * @returns {any} jsx
   * @memberof SignIn
   */
  render() {
    const {
      message, showProps, resetPass, success
    } = this.state;
    return (
      <section className="container mt-100 mb-100 ">
        <div className="fixed-top">
          {this.props.netReq === true ? (
            <Pace color="#e7b52c" height={2} />
          ) : null}
        </div>
        <Navbar className="bg-dark fixed-top" />
        {showProps && (
          <div className="alert alert-warning" role="alert">
            <strong>{message}</strong>
          </div>
        )}
        {success && (
          <div
            className="alert alert-success alert-dismissible fade show"
            role="alert"
          >
            <strong>New Password Saved!</strong> Login with new password
            <button
              type="button"
              className="close"
              data-dismiss="alert"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
        )}
        <div
          className="row p-0 catalog-wrapper justify-content-center"
          data-aos="zoom-out"
        >
          <div className="AuthInfo col-lg-7 col-sm-12 justify-content-center text-center ">
            <img
              src="https://res.cloudinary.com/emasys/image/upload/v1516439649/mR_2_jwnuce.png"
              alt="logo"
              width="200"
              height="200"
              className="mt-30"
              data-aos="flip-right"
              data-aos-delay="1000"
              data-dos-duration="1000"
            />
            <h1
              className="text-white"
              data-aos="fade-up"
              data-aos-duration="2000"
            >
              Welcome back!
            </h1>
            <h4
              className="mt-10 text-white mb-10 pr-50 pl-50"
              data-aos="fade-up"
              data-aos-duration="1000"
              data-dos-delay="1000"
            >
              We trust it's been an amazing experience for you so far... Let's
              continue to add spices to life.
            </h4>
          </div>
          <div className="col-lg-5 col-sm-8 pb-20 signin-form">
            {!resetPass && (
              <SignInForm
                handleSubmit={this.handleSubmit}
                state={this.state}
                clearError={this.clearError}
                resetForm={this.resetForm}
              />
            )}
            {resetPass && (
              <ResetPasswordForm
                state={this.state}
                onChange={this.onChange}
                emailChanged={this.emailChanged}
                resetPassword={this.resetPassword}
                generateToken={this.generateToken}
                resetForm={this.resetForm}
              />
            )}
          </div>
        </div>
      </section>
    );
  }
}

const mapStateToProps = state => ({
  signin: state.user,
  reset: state.user.reset,
  netReq: state.netReq
});

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators(
    {
      resetPassword,
      signIn,
      compareToken,
      sendToken,
      isLoading
    },
    dispatch
  )
});

SignIn.propTypes = {
  msg: PropTypes.string,
  reset: PropTypes.object,
  resetPassword: PropTypes.func,
  signin: PropTypes.object,
  signIn: PropTypes.func,
  match: PropTypes.object,
  location: PropTypes.object,
  compareToken: PropTypes.func,
  sendToken: PropTypes.func,
  netReq: PropTypes.bool
};

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);
