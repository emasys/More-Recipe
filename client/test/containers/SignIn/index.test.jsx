import React from 'react';
import { mount, shallow } from 'enzyme';
import Router from 'react-mock-router';

// Component
import { SignIn, mapStateToProps } from '../../../src/containers/SignIn/index';
import fakeStore from '../../__mocks__/fakeStore';

let wrapper = null;

jest.mock('../../../src/containers/validator/index');

jest.mock('../../../src/components/Navbar');

const props = {
  msg: 'you cannot view this page',
  reset: {},
  signin: fakeStore.user,
  signIn: jest.fn(),
  compareToken: jest.fn(),
  history: { push: jest.fn() },
  sendToken: jest.fn(),
  clearAuthInfo: jest.fn()
};
describe('Test suite for SignIn page', () => {
  describe('sign in', () => {
    beforeEach(() => {
      wrapper = mount(<Router>
        <SignIn {...props} />
      </Router>);
    });

    it('Should render without errors', () => {
      expect(wrapper.length).toEqual(1);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('h4.mt-10.text-white.mb-10.pr-50.pl-50.pb-20').text()).toMatch('We trust it has been an amazing experience for you so far... Keep adding spices to life!');
    });

    it('Should complete login form and submit', () => {
      const handleSubmit = jest.spyOn(SignIn.prototype, 'handleSubmit');
      // const onFocus = jest.spyOn(SignIn.prototype, 'onFocus');
      const clearError = jest.spyOn(SignIn.prototype, 'clearError');
      const app = mount(<Router><SignIn {...props} /></Router>);

      app
        .find("input[name='email']")
        .simulate('change', { target: { value: 'sample@example.com' } });
      app.find("input[name='pass']").simulate('focus');
      app
        .find("input[name='pass']")
        .simulate('change', { target: { value: 'p@ssw0rd' } });
      app.find('form').simulate('submit');

      expect(handleSubmit).toHaveBeenCalled();
      expect(clearError).toHaveBeenCalled();
    });

    it('Should complete reset password form and submit', () => {
      const generateToken = jest.spyOn(SignIn.prototype, 'generateToken');
      const onFocus = jest.spyOn(SignIn.prototype, 'onFocus');
      const onChange = jest.spyOn(SignIn.prototype, 'onChange');
      const onBlur = jest.spyOn(SignIn.prototype, 'onBlur');
      const resetPassword = jest.spyOn(SignIn.prototype, 'resetPassword');

      const app = mount(<Router><SignIn {...props} /></Router>);

      app.find('span#forgotPassword').simulate('click');
      expect(app).toMatchSnapshot();
      app
        .find("input[name='recoveryEmail']")
        .simulate('change', { target: { value: 'sample007@example.com' } });
      app.find('#sendToken').simulate('click');
      expect(generateToken).toHaveBeenCalled();
      app
        .find("input[name='newPassword']")
        .simulate('change', { target: { value: 'p@ssw0rd' } });
      app.find("input[name='newPassword']").simulate('focus');
      app.find("input[name='newPassword']").simulate('blur');
      expect(onBlur).toHaveBeenCalled();
      expect(onFocus).toHaveBeenCalled();
      app
        .find("input[name='confirmPassword']")
        .simulate('change', { target: { value: 'p@ssw0rd' } });
      app
        .find("input[name='token']")
        .simulate('change', { target: { value: '1234' } });
      expect(onChange).toHaveBeenCalled();
      app.find('form').simulate('submit');
      expect(resetPassword).toBeCalled();
    });

    it('should test mapStateToProps', () => {
      const initialState = {
        user: {},
        isLoading: false
      };
      expect(mapStateToProps(initialState).signin).toEqual({});
      expect(mapStateToProps(initialState).isLoading).toEqual(false);
    });
  });

  it('should call componentWillReceiveProps', () => {
    const app = shallow(<SignIn {...props} />);
    const instance = app.instance();
    const compWRP = jest.spyOn(instance, 'componentWillReceiveProps');
    const newProps = {
      user: {
        compareToken: { success: true },
        reset: {
          success: true,
          status: 'updated'
        }
      }
    };

    const nextProps = {
      user: {
        signIn: { tokenStatus: true }
      }
    };

    app.setProps(newProps);
    expect(compWRP).toHaveBeenCalled();
    expect(instance.state.tokenError).toBe('d-none');

    app.setProps(nextProps);
    expect(compWRP).toHaveBeenCalled();
  });
});
