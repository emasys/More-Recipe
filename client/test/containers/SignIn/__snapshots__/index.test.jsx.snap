// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for SignIn page sign in Should complete reset password form and submit 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <SignIn
    clearAuthInfo={[Function]}
    compareToken={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    msg="you cannot view this page"
    reset={Object {}}
    sendToken={[Function]}
    signIn={[Function]}
    signin={
      Object {
        "allUsers": Object {
          "success": true,
          "users": Array [
            Object {
              "avatar": null,
              "bio": null,
              "country": null,
              "createdAt": "2018-03-13T05:04:37.443Z",
              "email": "emasysnd@gmail.com",
              "firstName": null,
              "id": 1,
              "lastName": null,
              "moniker": "admin",
              "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
              "updatedAt": "2018-03-13T05:04:37.443Z",
            },
          ],
        },
        "authInfo": Object {
          "isLoggedIn": true,
          "userId": 1,
          "username": "admin",
        },
        "isLoggedIn": true,
        "message": null,
        "userInfo": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": "Nigeria",
            "email": "emasysnd@gmail.com",
            "firstName": "emmy",
            "id": 1,
            "lastName": "endy",
            "moniker": "admin",
          },
          "success": true,
        },
        "userProfile": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": null,
            "email": "emasysnd@gmail.com",
            "firstName": null,
            "id": 1,
            "lastName": null,
            "moniker": "admin",
          },
          "success": true,
        },
      }
    }
  >
    <section
      className="container mt-100 mb-100 "
    >
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={<DefaultCloseButton />}
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      >
        <div />
      </ToastContainer>
      <Connect(Navbar)
        className="bg-dark fixed-top"
      />
      <div
        className="row p-0 justify-content-center"
        data-aos="fade-up"
        data-aos-duration="1000"
      >
        <div
          className="catalog-wrapper col-lg-6 col-md-9 p-0"
        >
          <div
            className="AuthInfo col-12 text-center "
          >
            <img
              alt="logo"
              className="mt-30"
              data-aos="flip-right"
              data-aos-delay="1000"
              data-dos-duration="1000"
              height="200"
              src="https://res.cloudinary.com/emasys/image/upload/v1516439649/mR_2_jwnuce.png"
              width="200"
            />
            <h1
              className="text-white"
            >
              Welcome back!
            </h1>
            <h4
              className="mt-10 text-white mb-10 pr-50 pl-50 pb-20"
            >
              We trust it has been an amazing experience for you so far... Keep adding spices to life!
            </h4>
          </div>
          <div
            className="col-12 pb-20 signin-form"
          >
            <ResetPasswordForm
              generateToken={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              resetForm={[Function]}
              resetPassword={[Function]}
              state={
                Object {
                  "buttonName": "Send Token",
                  "confirmPassword": "",
                  "display": "d-none",
                  "email": "",
                  "hideForm": "d-none",
                  "message": "you cannot view this page",
                  "newPassword": "",
                  "password": "",
                  "recoveryEmail": "",
                  "resetPass": true,
                  "showErrMessage": "fade",
                  "showProps": true,
                  "showRubics": "d-none",
                  "success": false,
                  "tokenError": "d-none",
                }
              }
            >
              <form
                className=""
                name="form"
                onSubmit={[Function]}
              >
                <ul
                  className="form row"
                >
                  <li
                    className="col-lg-12 col-sm-12"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      className="col-lg-12 col-sm-12"
                      id="recoveryEmail"
                      name="recoveryEmail"
                      onChange={[Function]}
                      placeholder="example@example.com"
                      required={true}
                      type="email"
                      value=""
                    />
                    <div
                      className="text-danger d-none"
                      id="recoverEmail_error"
                    >
                      This email address is not in our records
                    </div>
                    <button
                      className="btn btn-info mt-5 btn-block btn-lg "
                      id="sendToken"
                      name="sendToken"
                      onClick={[Function]}
                    >
                      Send Token
                    </button>
                  </li>
                  <li
                    className="col-lg-12 col-sm-12 d-none d-none"
                  >
                    <label>
                      New Password
                    </label>
                    <input
                      className="col-lg-12 col-sm-12"
                      id="newPassword"
                      name="newPassword"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="min of 8 characters"
                      required={true}
                      type="password"
                      value=""
                    />
                    <div
                      className="rubics d-none"
                    >
                      <p
                        className="text-danger"
                        id="password_error"
                      />
                      <p
                        className="alpha"
                      >
                        Should contain at least one alphabet
                      </p>
                      <p
                        className="numbers"
                        id="check_numbers"
                      >
                        Should contain at least one number
                      </p>
                      <p
                        className="characters"
                      >
                        Should contain at least one special character
                      </p>
                      <p
                        className="minLength"
                      >
                        Should not be less than 8 characters
                      </p>
                      <p
                        className="maxLength"
                      >
                        Should not be greater than 20 characters
                      </p>
                    </div>
                  </li>
                  <li
                    className="col-lg-12 col-sm-12 d-none d-none"
                  >
                    <label>
                      Confirm New Password
                    </label>
                    <input
                      className="col-lg-12 col-sm-12"
                      id="confirmPassword"
                      name="confirmPassword"
                      onChange={[Function]}
                      placeholder="min of 8 characters"
                      required={true}
                      type="password"
                    />
                    <div
                      className="text-danger"
                      id="confirmPassword_error"
                    />
                  </li>
                  <li
                    className="col-lg-12 col-sm-12 d-none d-none"
                  >
                    <label>
                      Token
                    </label>
                    <input
                      className="col-lg-12 col-sm-12 "
                      name="token"
                      onFocus={[Function]}
                      placeholder="enter 4-digit token"
                      required={true}
                      type="text"
                    />
                    <div
                      className="text-danger d-none"
                      id="token_error"
                    >
                       Invalid Token
                    </div>
                  </li>
                  <li
                    className="col-lg-12 col-sm-12 d-none d-none"
                  >
                    <button
                      className="btn btn-dark btn-block btn-lg"
                      type="submit"
                    >
                      Save New Password
                    </button>
                  </li>
                  <li
                    className="col-12 text-center"
                  >
                    <span
                      className="text-info hovered"
                      onClick={[Function]}
                    >
                      Go back to sign in
                    </span>
                  </li>
                </ul>
              </form>
            </ResetPasswordForm>
          </div>
        </div>
      </div>
    </section>
  </SignIn>
</MockRouter>
`;

exports[`Test suite for SignIn page sign in Should render without errors 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <SignIn
    clearAuthInfo={[Function]}
    compareToken={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    msg="you cannot view this page"
    reset={Object {}}
    sendToken={[Function]}
    signIn={[Function]}
    signin={
      Object {
        "allUsers": Object {
          "success": true,
          "users": Array [
            Object {
              "avatar": null,
              "bio": null,
              "country": null,
              "createdAt": "2018-03-13T05:04:37.443Z",
              "email": "emasysnd@gmail.com",
              "firstName": null,
              "id": 1,
              "lastName": null,
              "moniker": "admin",
              "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
              "updatedAt": "2018-03-13T05:04:37.443Z",
            },
          ],
        },
        "authInfo": Object {
          "isLoggedIn": true,
          "userId": 1,
          "username": "admin",
        },
        "isLoggedIn": true,
        "message": null,
        "userInfo": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": "Nigeria",
            "email": "emasysnd@gmail.com",
            "firstName": "emmy",
            "id": 1,
            "lastName": "endy",
            "moniker": "admin",
          },
          "success": true,
        },
        "userProfile": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": null,
            "email": "emasysnd@gmail.com",
            "firstName": null,
            "id": 1,
            "lastName": null,
            "moniker": "admin",
          },
          "success": true,
        },
      }
    }
  >
    <section
      className="container mt-100 mb-100 "
    >
      <ToastContainer
        autoClose={5000}
        bodyClassName=""
        className={null}
        closeButton={<DefaultCloseButton />}
        closeOnClick={true}
        hideProgressBar={false}
        newestOnTop={false}
        pauseOnHover={true}
        position="top-right"
        progressClassName=""
        style={null}
        toastClassName=""
        transition={[Function]}
      >
        <div />
      </ToastContainer>
      <Connect(Navbar)
        className="bg-dark fixed-top"
      />
      <div
        className="row p-0 justify-content-center"
        data-aos="fade-up"
        data-aos-duration="1000"
      >
        <div
          className="catalog-wrapper col-lg-6 col-md-9 p-0"
        >
          <div
            className="AuthInfo col-12 text-center "
          >
            <img
              alt="logo"
              className="mt-30"
              data-aos="flip-right"
              data-aos-delay="1000"
              data-dos-duration="1000"
              height="200"
              src="https://res.cloudinary.com/emasys/image/upload/v1516439649/mR_2_jwnuce.png"
              width="200"
            />
            <h1
              className="text-white"
            >
              Welcome back!
            </h1>
            <h4
              className="mt-10 text-white mb-10 pr-50 pl-50 pb-20"
            >
              We trust it has been an amazing experience for you so far... Keep adding spices to life!
            </h4>
          </div>
          <div
            className="col-12 pb-20 signin-form"
          >
            <SignInForm
              clearError={[Function]}
              handleSubmit={[Function]}
              resetForm={[Function]}
              state={
                Object {
                  "buttonName": "Send Token",
                  "confirmPassword": "",
                  "display": "d-none",
                  "email": "",
                  "hideForm": "d-none",
                  "message": "you cannot view this page",
                  "newPassword": "",
                  "password": "",
                  "recoveryEmail": "",
                  "resetPass": false,
                  "showErrMessage": "fade",
                  "showProps": true,
                  "showRubics": "d-none",
                  "success": false,
                  "tokenError": "d-none",
                }
              }
            >
              <form
                className="form-ite"
                id="signin"
                onSubmit={[Function]}
              >
                <ul
                  className="form row"
                >
                  <li
                    className="col-lg-12 col-sm-12"
                  >
                    <label>
                      Email
                    </label>
                    <input
                      className="col-lg-12 col-sm-12"
                      id="inputEmail"
                      name="email"
                      onFocus={[Function]}
                      placeholder="example@example.com"
                      required={true}
                      type="email"
                    />
                  </li>
                  <li
                    className="col-lg-12 col-sm-12"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      className="col-lg-12 col-sm-12"
                      id="inputPassword"
                      maxLength="20"
                      name="pass"
                      onFocus={[Function]}
                      placeholder="**********"
                      required={true}
                      type="password"
                    />
                    <div
                      className="text-danger fade"
                      id="error"
                    >
                      Invalid email or password
                    </div>
                  </li>
                  <li
                    className="col-lg-12 col-sm-12 text-center"
                  >
                    <button
                      className="btn btn-dark btn-lg col-5"
                      type="submit"
                    >
                      Sign in
                    </button>
                     
                    <Link
                      className="btn btn-dark btn-lg col-5"
                      replace={false}
                      to="/signup"
                    >
                      <a
                        className="btn btn-dark btn-lg col-5"
                        onClick={[Function]}
                      >
                        Sign up
                      </a>
                    </Link>
                  </li>
                  <li
                    className="col-lg-12 col-sm-12 text-center"
                  >
                    <span
                      className="text-info hovered"
                      id="forgotPassword"
                      onClick={[Function]}
                    >
                      forgot password?
                    </span>
                  </li>
                </ul>
              </form>
            </SignInForm>
          </div>
        </div>
      </div>
    </section>
  </SignIn>
</MockRouter>
`;
